### PR TITLE
ci: merge docs builds

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

reduce duplication and so potential issues with missed updates

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca

### CODE REVIEW

File: .github/workflows/docs-build.yml [hunk 1]
```diff
@@ -85,86 +82,40 @@ jobs:
           mkdir -p $PYPI_CACHE_DIR  # in case cache was not hit
           ls -lh $PYPI_CACHE_DIR
           mkdir -p pypi_pkgs  # in case template is not pulled
-          pip install .[extra,cloud,ui] -U -r requirements/${{ matrix.pkg-name }}/docs.txt \
+          pip install .[all] -U -r requirements/${{ matrix.pkg-name }}/docs.txt \
```
Comment (carmocca) [inline .github/workflows/docs-build.yml:85]: This will include testing and example deps. I suggest reverting this change

**Review by carmocca — APPROVED** [review]

File: .github/workflows/docs-build.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/docs-build.yml:85]: I think it is fine...

Comment (awaelchli) [reply .github/workflows/docs-build.yml:85]: @Borda what does "all" include in addition to the previous ones?

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/docs-build.yml (hunk 1 above)
Comment (awaelchli) [reply .github/workflows/docs-build.yml:85]: Where can I find the mapping from these tags to dependencies?

Comment (Borda) [reply .github/workflows/docs-build.yml:87]: then all `*-all` are aggregated as `all`

**Pull request merged**